AltV.Community.MValueAdapters.Generators/Converters/Other/EnumConverter.cs

```diff
@@ -0,0 +1,45 @@
+using System.Text;
+using AltV.Community.MValueAdapters.Generators.Models;
+
+namespace AltV.Community.MValueAdapters.Generators.Converters;
+
+internal class EnumConverter : BaseConverter
+{
+    protected override void GenerateItemWriteCode(StringBuilder stringBuilder, ref int indentation, MValueClassInfo classInfo, MValuePropertyInfo propertyInfo)
+    {
+        stringBuilder.AppendLine(indentation, $"writer.Value(({propertyInfo.UnderlyingEnumTypeName})value.{propertyInfo.Name});");
```

Comment:
If the underlying type is one of the number types, the value should be casted one more time to the double type. It would be enum &rarr; underlying type &rarr; double, or simply enum &rarr; double.

Reply:
I've adapted the code to always make numbers (double) when generating `ToMValue` code.
It got a bit messy, so I cleaned up stuff.

Feel free to have a look